Source/items.cpp

```diff
@@ -3198,6 +3198,27 @@ void DoRecharge(int pnum, int cii)
 		SetCursor_(CURSOR_HAND);
 }
 
+#ifdef HELLFIRE
+void DoOil(int pnum, int cii)
+{
+	PlayerStruct *p = &plr[pnum];
+
+	if (cii >= 7 || cii == 0 || (cii > 3 && cii <= 6)) {

```

Comment:
```suggestion
	if (cii >= NUM_INVLOC || cii == INVLOC_HEAD || (cii > INVLOC_AMULET && cii <= INVLOC_CHEST)) {
```

Reply:
just wondering, why is >= 7 check there? that will cause OOB for InvBody

Reply:
or were they using the OOB on purpose to access inventory? :D
`	ItemStruct InvBody[NUM_INVLOC];
	ItemStruct InvList[NUM_INV_GRID_ELEM];`
lol

Reply:
probably